image_tools_py/src/cam2image_py.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# System imports
+import sys
+import argparse
+
+# ROS2 imports
+import rclpy
+from rclpy.qos import qos_profile_sensor_data
+import std_msgs.msg
+import sensor_msgs.msg
+
+# OpenCV imports
+import cv2
+
+# Local imports
+import burger_py
+import image_qos_profile
+
+is_flipped = False
+
+
+def flip_image_cb(msg):
+    global is_flipped
+    is_flipped = msg.data
+
+    output = 'off'
+    if is_flipped:
+        output = 'on'
+
+    print('Set flip mode to: ' + output)
+
+
+# Convert an OpenCV matrix encoding type to a string format recognized by
+# sensor_msgs::Image
+def mat2encoding(frame):
+    encoding = ''
+
+    encodings = {1: 'mono', 3: 'bgr', 4: 'rgba'}
+    for channels, prefix in encodings.items():
+        if frame.shape[2] == channels:
+            encoding += prefix
```

Comment:
For this particular loop, yes.  However, I prefer to make it more symmetric with the loop below; that way, if we ever somehow added another loop and someone copy-n-pasted from above, it would do the right thing.